R/github.R

```diff
@@ -23,16 +23,51 @@ github_response <- function(req) {
   parsed
 }
 
-github_GET <- function(path, ..., pat = github_pat()) {
+github_GET <- function(path, ..., pat = github_pat(),
+                       host = "api.github.com") {
+
+  url <- paste0("https://", host, "/")
+
+  # if the host contains a path, i.e. "github.hostname.com/api/v3",
```

Comment:
I do this because the endpoint prefix for enterprise Github is different from github.com.

enterprise: `http(s)://hostname/api/v3/` [ref](https://developer.github.com/v3/enterprise/)
github.com: `https://api.github.com` [ref](https://developer.github.com/v3/)

Of course, this does not mean that stuffing everything into `host` is the right way to do it :)

Just to spitball on a possible paths forward, I could use two args (with defaults): `host = "https://api.github.com"` and `path_prefix = NULL`. To use enterprise github, one could specify 
`host = "https://github.hostname.com"` and `path_prefix = "api/v3"`.

Please let me know what you think and I'll change things accordingly.